store/set.js

```diff
@@ -1,51 +1,131 @@
-const setIdFromUri = (uri) => uri.split('/item').pop();
-
 export const state = () => ({
   likesId: null,
-  liked: []
+  likedItems: [],
+  likedItemIds: [],
+  active: null,
+  creations: []
 });
 
 export const mutations = {
   setLikesId(state, value) {
     state.likesId = value;
   },
-  setLikedItems(state, likesArray) {
-    state.liked = likesArray;
+  setLikedItems(state, value) {
+    state.likedItems = value;
+    state.likedItemIds = value.map(item => item.id);
+  },
+  like(state, itemId) {
+    state.likedItemIds.push(itemId);
+  },
+  unlike(state, itemId) {
+    state.likedItemIds.splice(state.likedItemIds.indexOf(itemId), 1);
```

Comment:
Unlike should also remove the item from likedItems. Currently the grid still shows the item if you unlike it while you are on the likes page.